src/agent/sqlite_manager/include/sqlite_manager.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <SQLiteCpp/SQLiteCpp.h>
+#include <SQLiteCpp/VariadicBind.h>
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace sqlite_manager
+{
+    const std::map<int, std::string> mapColTypeString {
+        {SQLite::INTEGER, "INTEGER"},
+        {SQLite::TEXT, "TEXT"},
+        {SQLite::FLOAT, "REAL"},
+        {SQLite::BLOB, "BLOB"},
+        {SQLite::Null, "Null"},
+    };
+
+    class Col
+    {
+    public:
+        Col()
+            : m_type(SQLite::Null)
+            , m_notNull(false)
+            , m_autoIncrement(false)
+            , m_primaryKey(false)
+        {
+        }
+
+        Col(const std::string& name,
+            const int type,
+            const bool notNull,
+            const bool autoIncr,
+            const bool primary = false)
+            : m_name(name)
+            , m_type(type)
+            , m_notNull(notNull)
+            , m_autoIncrement(autoIncr)
+            , m_primaryKey(primary)
+        {
+        }
+
+        Col(const std::string& name, const int type, const std::string& value)
+            : m_name(name)
+            , m_type(type)
+            , m_value(value)
+        {
+        }
+
+        std::string m_name;
+        int m_type;
+        bool m_notNull;
+        bool m_autoIncrement;
+        bool m_primaryKey;
+        std::string m_value;
+    };
+
+    class SQLiteManager
+    {
+    public:
+        SQLiteManager(const std::string& dbName);
+        ~SQLiteManager() {};
+
+        int CreateTable(const std::string& tableName, const std::vector<Col>& cols);
+        int Insert(const std::string& tableName, const std::vector<Col>& cols);
+        int GetCount(const std::string& tableName);
+        int ExecuteSQL(const std::string& queryString);
+        int
```

Comment:
As we talk, the return function should be the value selected.

Reply:
By doing this, I think a single function will be necessary to select a single or multiple values.

Reply:
Done. The same function now returns a vector of rows, which in turn are vectors of Columns.